.github/workflows/cypress.yml

```diff
@@ -0,0 +1,128 @@
+name: Cypress CI Workflow
+
+on:
+  push:
+    branches:
+      - feat/user-role-permissions
+      - feature/Automated-e2e-Tests
+      - ft/add-device-test
+      - feature/add-meter-read
+  pull_request:
+    branches:
+      - ft/register-new-account
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    env:
+      MODE: dev
+      MNEMONIC: "shine ability artefact river cabin copy blind hello gift hospital tenant region"
+      WEB3: https://volta-rpc.energyweb.org
+      UI_BASE_URL: http://localhost:4200
+      INFLUXDB_URL: http://localhost:8086
+      INFLUXDB_TOKEN: admin:admin
+      INFLUXDB_BUCKET: energy/autogen
+      ISSUER_PRIVATE_KEY: 4773097f508a6edd206150766ee25cb5abc1635f28f9cfd27e144872e89dc9c9
+      DREC_BLOCKCHAIN_ADDRESS: "0xDeC1488838C65E355A9336eE8DdFc32665dB9222"
+      CERTIFICATE_QUEUE_DELAY: 60000
+      BACKEND_PORT: 3040
+      DB_HOST: localhost
+      DB_PORT: 5432
+      DB_USERNAME: postgres
+      DB_PASSWORD: postgres
+      DB_DATABASE: origin
+      REACT_APP_BACKEND_URL: http://localhost:3040
+      ADMIN_EMAIL: drec@energy.org
+      ADMIN_PASSWORD: 1234Rc
+
+    steps:
+      # Step 1: Checkout Repository
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      # Step 2: Install pnpm and Rush globally
+      - name: Install pnpm and Rush globally
+        run: |
+          npm install -g pnpm @microsoft/rush
+
+      # Step 3: Install System Dependencies
+      - name: Install System Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y influxdb-client docker-compose postgresql-client netcat-openbsd curl
+
+      # Step 4: Setup Node.js
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.14.0
+          cache: 'npm'
+
+      # Step 5: Clone Backend Repository
+      - name: Clone Backend Repository
+        run: |
+          git clone  https://github.com/d-rec/drec-origin.git projects/drec-origin
```

Comment:
can you define the branch you clone.

- If we are running the job on dev, we should clone the dev version of the api
- If we are running the job on staging, we should also clone the staging version of the api
- If we are running the job on production, we should also make sure to clone the production version of the api